resource_cabot_check_graphite.go

```diff
@@ -73,19 +75,25 @@ func resourceCabotCheckGraphiteRead(d *schema.ResourceData, meta interface{}) er
 	client := meta.(*cabot.Client)
 
 	id, _ := strconv.Atoi(d.Id())
-	check, err := client.GraphiteChecks.Get(id)
-	if err != nil {
-		return err
-	}
 
-	d.SetId(strconv.Itoa(check.ID))
-	setResourceDataForStatusCheck(d, check.StatusCheck)
-	d.Set("metric", check.Metric)
-	d.Set("check_type", check.CheckType)
-	d.Set("value", check.Value)
-	d.Set("expected_num_hosts", check.ExpectedNumHosts)
-	d.Set("allowed_num_failures", check.AllowedNumFailures)
-	return nil
+	return resource.Retry(DEFAULT_RETRY_TIMEOUT, func() *resource.RetryError {
+		time.Sleep(DEFAULT_GRACE_PERIOD)
```

Comment:
Ideally, it'll be nice to sleep on the start of the first failure. Not a huge fan on sleeping before execution or before returning an error.